### DISCUSSION TITLE
Planificación Externa en Java

### DISCUSSION THREAD

**Discussion by Gdepablo** [dds-jv/foro#378] [Consultas de Arquitectura] [answered]
Buenas, andaba reviendo el video de la clase después de la clase de repaso de hoy y la verdad es que no me queda claro cómo hace Roli para ejecutar la 'sugerencia' de QMP6 utilizando planificación externa. Entriendo que hay que agregarle un main pero no entiendo por ejemplo, cómo se definiría la periocidad con la que se ejecuta el evento sin usar el Timer de Java que es planifcación interna (y básicamente en la clase de hoy dijeron que en el 99% de los casos, planificacion interna = lava). La cuestión es que quería implementarlo para el requisito de las notificaciones que dado un tiempo yo tengo que recordarle cuando haya pasado ese tiempo.
Cuál sería la alternativa a utilizar timer de Java? No entiendo.

**Comment by gastonprieto** [comment]
Hola @Gdepablo 

La planificación externa, es un programa de tu sistema operativo o arquitectura al cual le decis quiero que ejecutes este X ejecutable en estos horarios (o en una expresión de tipo cron), esto es un ejemplo de configuración

```
# Por ejemplo corre un ls
# a las 5 a.m cada semana:
# 0 5 * * 1 ls
```
Pagina que traduce expresión de tipo cron a humano: https://crontab.guru/#0_5_*_*_1

Ahora, como unis esto con Java? Necesitas un ejecutable.

Para armar un ejecutable necesitas en Java una clase que tenga un `main()` y haga lo que vos queres hacer. En el caso de lo que veiamos en QMP6 es mandar sugerencias, pero podría ser lo que necesites.

Ahora que en java tenes un proyecto maven que tiene un main podes generar un binario, para esto maven te da una tarea que es package, y con eso tenes un .jar que puedas poner que se ejecute en el planificador externo que corresponda.

Se entendio un poco mas ?

**Comment by Gdepablo** [comment, marked as answer]
Basándome en el  QMP6 es que yo traté de usar el mismo plugin, pero no me lo tomaba el POM. 
Aparte de eso, no entiendo, o sea, en el main, ¿cómo hacés que te ejecute el cron? Eso fue lo que no me quedó claro del video, porque en el QMP6, lo ejecuta desde la terminal de Linux, y no como un método adentro del código en java.
No sé si me explico

**Reply by gastonprieto** [reply]
el cron **no** se ejecuta desde java, eso es lo primero a entender en la planificación externa, es externa a tu programa. O sea hay otro programa que se encarga de ejecutar al tuyo.

Es como si le dijeras a algun conocido, por favor a las 9am ejecutame este ejecutable (en este caso seria el ejecutable de java).

El main es el metodo que entiende la maquina virtual de Java q tiene que ejecutar y dispararía todas las acciones de tu programa. Es como el main(...) de C

**Reply by gastonprieto** [reply]
@Gdepablo quedo mas o menos ahora la idea?

**Reply by Gdepablo** [reply]
Buenas, perdón la demora.
Me quedó claro, pero la duda que me surgía en realidad es, por ejemplo, en el parcial, planteamos el código y sacamos una flechita que diga "Esto lo hacemos con planificación externa en Java" y ya está, no hay que codear ningún paso adicional ni nada por el estilo?
No sé si me explico

**Reply by gastonprieto** [reply]
claro, aunque diria solo "esto se llama con planificación externa", no en Java porque la planificación externa no es de Java
